### PR TITLE
Moving from scala 2 to scala 3

### PR DESCRIPTION
**Pull request by eloots**

### CODE REVIEW

**Comment by eloots** [comment]
Note that this missing the link to the CMT tech talk (not yet available, but that should change soon)

File: posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ifdef::backend-html5[]
+:in-between-width: width='85%'
+:half-width: width='50%'
+:half-size:
+:thumbnail: width='60'
+endif::[]
+
+== Moving From Scala 2 to Scala 3
+
+After some early peeks at https://dotty.epfl.ch[the Dotty website] on a few occasions in 2019, I started looking at the https://dotty.epfl.ch/docs/reference/overview.html[Dotty reference documentation] in earnest after reading the release notes of the https://dotty.epfl.ch/blog/2019/12/20/21th-dotty-milestone-release.html[0.21.0-RC1 Dotty release]. That release marked an important milestone, namely, the declaration of Dotty as feature-complete. After reading the article https://www.scala-lang.org/2019/12/18/road-to-scala-3.html[Scala 2 roadmap update: The Road to Scala 3] that was referenced in the release notes, I had all I needed to test some ideas with Dotty's new Union Types. You may have read the full report on this in https://www.lunatech.com/2020/02/using-dotty-union-types-with-akka-typed[part I] and https://www.lunatech.com/2020/02/using-dotty-union-types-with-akka-typed-part-ii[part II] of the ***Using Dotty Union types with Akka Typed*** articles.
```
Comment (ckipp01) [inline posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc:15]:
```suggestion
After some early peeks at https://dotty.epfl.ch[the Dotty website] on a few occasions in 2019, I started looking more closely at the https://dotty.epfl.ch/docs/reference/overview.html[Dotty reference documentation] after reading the release notes of the https://dotty.epfl.ch/blog/2019/12/20/21th-dotty-milestone-release.html[0.21.0-RC1 Dotty release]. That release marked an important milestone, namely, the announcement of Dotty as feature-complete. After reading the article https://www.scala-lang.org/2019/12/18/road-to-scala-3.html[Scala 2 roadmap update: The Road to Scala 3], which was referenced in the release notes, I had all I needed to test some ideas with Dotty's new Union Types. You may have read the full report on this in https://www.lunatech.com/2020/02/using-dotty-union-types-with-akka-typed[part I] and https://www.lunatech.com/2020/02/using-dotty-union-types-with-akka-typed-part-ii[part II] of the ***Using Dotty Union types with Akka Typed*** articles.
```

File: posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+Pretty exciting stuff I still think, and I'm sure that this approach will be adopted in a future version of the Akka toolkit as an alternative to the rather convoluted way of dealing with responses to messages in Akka Typed (something which is necessary because of some limitations of the Scala 2 compiler).
```
Comment (ckipp01) [inline posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc:17]: I'd recommend changing this around a bit and not starting the sentence with "Pretty exciting stuff I still think", which reads a bit choppy. I'd recommend still stating what you did about how the Akka toolkit may adopt this as an alternative to the rather convulated current way of doing it, and then say something along the lines of "While I'm still excited about this, I also started wondering about what to do next...", which I think is a natural segue.

File: posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+So, with that behind me, I was wondering about what to do next... Now, I've always been very keen on learning new stuff, but even more on sharing knowledge with others. I've taught many technical courses and [co-]developed different courses on Scala and Akka while at Lightbend. So, a little less than two months ago, I decided to start working on a new course that would target existing Scala 2 developers and that would help them in the move to Scala 3. The general idea I started from was to take an existing Scala 2 application, select a number of key features in Dotty, and apply these to the application in a step-by-step fashion. The end goal was to show how these features and improve and reduce the complexity in the original code.
```
Comment (ckipp01) [inline posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc:19]:
```suggestion
So, with that behind me, I was wondering about what to do next... Now, I've always been very keen on learning new stuff, but even more on sharing knowledge with others. I've taught many technical courses and [co-]developed different courses on Scala and Akka while at Lightbend. So, a little less than two months ago, I decided to start working on a new course that would target existing Scala 2 developers and that would help them with the move to Scala 3. The general idea I started from was to take an existing Scala 2 application, select a number of key features in Dotty, and apply these to the application in a step-by-step fashion. The end goal was to show how these features improve and reduce the complexity in the original code.
```

File: posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+When I launched the idea about this new course at Lunatech, I got two enthusiastic colleagues, _Neil Benn_ and _Robin Raju_, to join me in the effort. So, after about 2 months in elapsed time, we delivered the first edition of the two-day _"Moving from Scala 2 to Scala 3"_ course to a bunch of colleagues.
```
Comment (ckipp01) [inline posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc:21]:
```suggestion
When I floated the idea about this new course at Lunatech, I got two enthusiastic colleagues, _Neil Benn_ and _Robin Raju_, to join me in the effort. After about 2 months, we delivered the first edition of the two-day _"Moving from Scala 2 to Scala 3"_ course to a group of colleagues.
```

File: posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+Between the start of the project and the first delivery, the alpha version of the course was kindly and thoroughly reviewed by members of the Scala Center. We received a lot of very useful feedback which resulted in a series of corresponding updates to the course materials. Speaking of which: We have a series of 11 exercises, a course slide deck with approximately 140 slides, and a collection of code-snippets that can be used to demonstrate various features of the Dotty language treated in the course.
```
Comment (ckipp01) [inline posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc:23]:
```suggestion
Between the start of the project and the first delivery, the alpha version of the course was kindly and thoroughly reviewed by members of the Scala Center. We received a lot of very useful feedback which resulted in a series of corresponding updates to the course materials. Speaking of which: We have a series of 11 exercises, a course slide deck with approximately 140 slides, and a collection of code-snippets that can be used to demonstrate various features of the Dotty language covered in the course.
```

File: posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+Very important to note is that all these materials are open-sourced! You can find them in https://github.com/lunatech-labs/lunatech-scala-2-to-scala3-course[this public repository] on Github. This means that you can benefit from this at no extra cost but the effort you put into studying the course materials! _(Of course, should you wish to have this course taught to you in-class, ping us —although that will not be free of charge)_.
```
Comment (ckipp01) [inline posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc:25]: I'd recommend the changing the last part about "-although that will not be free of charge". I think that's sort of implied, and just saying something about  "should you wish to have this course taught to you in-class, feel free to reach out to us and we're happy to talk!". I think this reads a bit better.

File: posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+== What's next...
+
+So, _"What's next?"_, you may ask. Well, in fact, quite a bit... Here's list of things that are in the works:
+
+* We plan to keep the course current with the release candidates of Dotty. At the time of writing, Dotty is at release 0.24.0-RC1 with 0.25 around the corner.
+* Robin is working on implementing CI/CD on the project. When that is done, we will automatically release so-called _"studentified"_  and _"linearized"_ versions of the course.
```
Comment (ckipp01) [inline posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc:32]:
```suggestion
* Robin is working on implementing CI/CD on the project. When that is done, we will automatically release the so-called _"studentified"_  and _"linearized"_ versions of the course.
```

File: posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+In a nutshell, a _"studentified"_ version of the course is an artefact generated from the https://github.com/lunatech-labs/lunatech-scala-2-to-scala3-course/tree/exercises-master/exercises[course master repository]. This artefact is a self-contained sbt project that has special functionality built-in to provide a friendly learning experience. A student can navigate between different exercises and thereby pull in relevant test code automatically, read exercise instructions via a manual page command, save and restore the state of the code, and _"pull"_ the solution if required.
```
Comment (ckipp01) [inline posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc:34]: Today I learned *artifact* can also be spelled *artefact*, and it's still technically correct. 🤔

Comment (ckipp01) [inline posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc:34]:
```suggestion
In a nutshell, a _"studentified"_ version of the course is an artefact generated from the https://github.com/lunatech-labs/lunatech-scala-2-to-scala3-course/tree/exercises-master/exercises[course master repository]. This artefact is a self-contained sbt project that has special functionality built-in to provide a friendly learning experience. A student can navigate between different exercises, pull in relevant test code automatically, read exercise instructions via a manual page command, save and restore the state of the code, and _"pull"_ the solution if required.
```

File: posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+A _"linearized"_ version of the course is a _git_ repository where every exercise in the course has a corresponding commit in the repo. As such, it can be used to quickly inspect the changes between subsequent exercises. This artefact also serves another purpose: it is an essential component in a workflow to evolve the code in exercises using _git interactive rebasing_.
+
+The transformation to _"studentified"_  and _"linearized"_ artefacts is done by the https://github.com/lightbend/course-management-tools[Course Management Tools] _(CMT)_ that I developed at https://www.lightbend.com[Lightbend].
+
+Neil, Robin and myself have a tech talk to colleagues last week about _CMT_. You can view a recording of this talk https://youtube.com[here].
```
Comment (ckipp01) [inline posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc:40]:
```suggestion
Neil, Robin and myself had a tech talk with colleagues last week about the _CMT_. You can view a recording of this talk https://youtube.com[here].
```

File: posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc (hunk 6 above)
Comment (eloots) [reply posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc:25]: Agreed and a great suggestion!

File: posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc (hunk 8 above)
Comment (eloots) [reply posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc:34]: I guess this is a US- versus British English thing? My spelling checker always wants to correct it to _artefact_. I also didn't know the other spelling was valid until it was suggested to me...

**Comment by eloots** [comment]
@ckipp01 
> I'd really recommend cutting the text width down.

I had totally forgotten about that one. I use MacDown to edit the source. It doesn't render AsciiDoc correctly, but at least I have a decent spelling checker. For the rendering, I use asciidoctor to generate html which I can inspect using a browser.

File: posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc (hunk 2 above)
Comment (eloots) [reply posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc:17]: I'm removing the "Pretty exciting..." part and joining this paragraph with the previous one.

File: posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc (hunk 9 above)
Comment (eloots) [reply posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc]: Changed as per your suggestion

**Pull request merged**